src/docs/guides/23-how-to-use-custom-pricing/index.md

```diff
@@ -0,0 +1,793 @@
+any changes as they can be combined with custom pricing.
+
+## 2. Listing extended data
+
+Custom pricing can be based on a lot of variables but one practical way
+to build it is to base it on information stored as extended data in
+listings. See the
+[How to extend listing data in FTW](/guides/how-to-extend-listing-data-in-ftw/)
+guide to read how to extend the listing data model with extended data.
+
+## 3. Change current booking model to use custom pricing
+
+The first step to do in FTW is to modify the current booking creation so
+that custom pricing is used instead of the nightly or daily pricing. For
+custom pricing, the booking is not constructed in the API based on
+`bookingStart` and `bookinEnd` but it's based on line items that the
+client sends in the booking initiation request.
+
+Instead of the just listing ID, now we're going to pass the actual line
+items in the transaction initiate request. To achieve this
+`CheckoutPage` has to be modified a bit. Add the following lines after
+the current imports at the top of the `CheckoutPage.js` file:
+
+```js
+import { nightsBetween, daysBetween } from '../../util/dates';
+import { types as sdkTypes } from '../../util/sdkLoader';
+
+const { Money } = sdkTypes;
+```
+
+Then add the following method to the `CheckoutPageComponent` class:
+
+```js
+/**
+ * Constructs a request params object that can be used when creating bookings
+ * using custom pricing.
+ * @param {} params An object that contains bookingStart, bookingEnd and listing
+ * @return a params object for custom pricing bookings
+ */
+customPricingParams(params) {
+  const { bookingStart, bookingEnd, listing, ...rest } = params;
+  const { amount, currency } = listing.attributes.price;
+
+  const unitType = config.bookingUnitType;
+  const isNightly = unitType === LINE_ITEM_NIGHT;
+
+  const quantity = isNightly
+    ? nightsBetween(bookingStart, bookingEnd)
+    : daysBetween(bookingStart, bookingEnd);
+
+  const bookingLineItem = {
+    code: unitType,
+    unitPrice: new Money(amount, currency),
+    quantity,
+  };
+
+  const lineItems = [bookingLineItem];
+
+  return {
+    listingId: listing.id,
+    bookingStart,
+    bookingEnd,
+    lineItems,
+    ...rest,
+  };
+}
+```
+
+When it's in place the method can be used to resolve new kind of request
+params.
+
+In the `loadInitialData` method:
+
+```js
+fetchSpeculatedTransaction(
+  this.customPricingParams({
+    listing,
+    bookingStart: bookingStartForAPI,
+    bookingEnd: bookingEndForAPI,
+  })
+);
+```
+
+And in `handleSubmit`:
+
+```js
+const requestParams = this.customPricingParams({
+  listing: this.state.pageData.listing,
+  cardToken,
+  bookingStart: speculatedTransaction.booking.attributes.start,
+  bookingEnd: speculatedTransaction.booking.attributes.end,
+});
+```
+
+## 4. Add cleaning fee to the listing page UI
+
+Next step is to modify the listing page UI so that a user can add the
+cleaning fee to the order. This guide expects that the cleaning fee
```

Comment:
I think it would be better to see the target image here. This is a long chapter and, at least for me, it would feel better to see the objective before diving to actual code examples.